spring-integration-amqp/src/main/java/org/springframework/integration/amqp/support/DefaultAmqpHeaderMapper.java

```diff
@@ -359,7 +360,8 @@ private String extractContentTypeAsString(Map<String, Object> headers) {
 		if (contentType != null) {
 			String contentTypeClassName = contentType.getClass().getName();
 
-			if (contentTypeClassName.equals("org.springframework.http.MediaType")) { // see INT-2713
+			if (contentType instanceof MimeType ||
```

Comment:
Why don't we stick with the `className` in the same way as `MediaType` - just in case of classloader issues ?

I realize `MimeType` is in spring core (Vs. spring-web for `MediaType`) and the reason for the other name was to avoid the hard dependency on spring-web but it seems it wouldn't hurt anything.

OTOH, why don't we simply do this (for 4.2 only)...

	if (contentType != null) {
		String contentTypeClassName = contentType.getClass().getName();

		contentTypeStringValue = contentType.toString();

		if (!(contentType instanceof MimeType
				|| contentTypeClassName.equals("org.springframework.http.MediaType")
				|| contentType instanceof String)) { // see INT-2713
			if (logger.isDebugEnabled()) {
				logger.warn("Header '" + AmqpHeaders.CONTENT_TYPE +
						"' may have an unexpected value since it is not of expected type [" + contentTypeClassName + "]");
			}
		}
	}